test/opaque_closure.jl

```diff
@@ -230,3 +230,26 @@ call_global_opaque_closure() = GLOBAL_OPAQUE_CLOSURE()
 let oc = @opaque a->sin(a)
     @test length(code_typed(oc, (Int,))) == 1
 end
+
+# constructing an opaque closure from IRCode
+using Core.Compiler: IRCode
+using Core: CodeInfo
+
+function OC(ir::IRCode, env...)
+    src = ccall(:jl_new_code_info_uninit, Ref{CodeInfo}, ());
+    src.slotflags = UInt8[]
+    nargs = length(ir.argtypes)
```

Comment:
`length(ir.argtypes)` doesn't necessarily correspond to the # of arguments of method from which this `ir` is constructed. IIUC that information disappeared within `CodeInfo`/`IRCode`. I think this entry point should take `nargs::Int` manually?